src/app/teacher/create-tag-dialog/create-tag-dialog.component.ts

```diff
@@ -45,14 +62,31 @@ export class CreateTagDialogComponent implements OnInit {
     this.subscriptions.unsubscribe();
   }
 
+  private createUniqueTagValidator(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      return this.tags != null &&
```

Comment:
Do we need this null check if we've already initialized it to an empty array?
```suggestion
      return 
```

Reply:
I added the null check because for some reason the validator is called before this.tags is initialized.